src/wallet/walletdb.h

```diff
@@ -112,11 +112,19 @@ class CKeyMetadata
     }
 };
 
-/** Access to the wallet database */
-class CWalletDB : public CDB
+/** Access to the wallet database.
+ * This should really be named CWalletDBBatch, as it represents a single transaction at the
+ * database. It will be committed when the object goes out of scope.
+ * Optionally (on by default) it will flush to disk as well.
+ */
+class CWalletDB
 {
 public:
-    CWalletDB(const std::string& strFilename, const char* pszMode = "r+", bool _fFlushOnClose = true) : CDB(strFilename, pszMode, _fFlushOnClose)
+    /** Takes a std::unique_ptr reference to avoid having to use
```

Comment:
This unique_ptr here bothers me. The interface shouldn't care what kind of pointer is used. I think I'm going to change this to take a `CWalletDBWrapper&`, then replace the call sites with `CWalletDB(*dbw)`. That's better than `CWalletDB(dbw.get())` at least.